app/policies/message_policy.rb

```diff
@@ -0,0 +1,23 @@
+class MessagePolicy < ApplicationPolicy
+  def index?
+    false
+  end
+
+  def create?
+    user.present?
+  end
+
+  def update?
+    user.present? && record.user == user
+  end
+
+  def destroy?
+    update?
+  end
+
+  class Scope < Scope
+    def resolve
+      scope.all
+    end
+  end
```

Comment:
If you pundit scope is same across all the policies, then you might as well not define it at all - it will default to defined in `ApplicationPolicy::Scope`.